### ISSUE TITLE
[SCREENREADER] Recommend reviewing language for error messages to improve descriptiveness

### ISSUE THREAD

**Issue by aeltarhoni** [department-of-veterans-affairs/va.gov-team#1711]
@1Copenut commented on [Mon Sep 24 2018](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/13536)

## Issue

Some of the error messages we are providing, especially for empty fields on the `blur` event firing, are general in their description. I'd like to review these and see if the content team has any suggestions for more specific errors. I have reviewed this with the VA 508 office and my contact agrees this is a nice enhancement, not a defect.

## Acceptance Criteria

As a screen reader user, I'd like to hear more specific error messages read out, that describe the type of error (which we are doing now) and which field it is related to.

## Environment

* Windows XP SP1
* IE11
* JAWS

---

![screen shot 2018-09-24 at 11 26 39 am](https://user-images.githubusercontent.com/934879/45965213-88ea2200-bfed-11e8-9189-51e65b8c500e.png)

---

@cehsu commented on [Tue Sep 25 2018](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/13536#issuecomment-424434043)

It seems like we will want to update these for all of our form fields. @erikphansen have you already created a ticket on the USFS repo regarding a similar issue?

---

@peggygannon commented on [Mon Feb 11 2019](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/13536#issuecomment-462551265)

@djmunoz1 @1Copenut @ Hi all, here are my content recommendations for updating the 526 error messages for 508 accessibility: 

- Please enter your 10-digit phone number (with or without dashes).
- Please enter your email address using this format: X@X.com.
- Please enter your street address.
- Please enter your city.

**Comment by smoonpeyes** [comment]
@Mottie It looks like @peggygannon has provided content for the error messages in the previous comment. I don't think there is any additional UX work needed.

**Comment by Mottie** [comment]
Ok, sorry about that!

**Comment by Mottie** [comment]
Actually @smoonpeyes I found multiple places where "Please provide a response" was used, including first name & last name; I'll follow the above convention for now, then tag you & Kevin once I submit the PR.

**Comment by Mottie** [comment]
@peggygannon Here's an updated screenshot of the current page showing all the error messages that need to be addressed:

<details>
<summary>526EZ Contact Information</summary>


![staging va gov_disability_file-disability-claim-form-21-526ez_contact-information](https://user-images.githubusercontent.com/136959/67951318-4401c900-fbb9-11e9-923d-4b6848fd5145.png)
</details>

**Comment by peggygannon** [comment]
Hi, @Mottie Here are my recommendations for the 526EZ in-line messages. Please let me know if you have any questions. (the pattern for these messages should be "Please enter your [required information]

Please enter your phone number
Please enter your email address
Please enter your street address
Please enter your state
Please enter your postal code
Please enter your start date

**Comment by Mottie** [comment]
Sorry, I should have been more clear. There are two types of errors that need to be addressed:

1. Error shown when the input is empty (required error message),
    e.g. "Please enter your phone number".
2. Error shown when the user doesn't follow the input pattern (pattern match error message)
    e.g. "Please enter a 10-digit phone number (with or without dashes)".

Also, I think these should be applied globally, not just to form 526EZ. I can update my pull request to increase coverage, if needed.

**Comment by peggygannon** [comment]
hi, @Mottie here are two messages for when a user doesn't follow the pattern input. Can you give me other examples of form inputs where a user isn't following the pattern? ie, are we also seeing this error for addresses? SSN? 

phone number: 
Please enter your 10-digit phone number (with or without dashes).

email address: 
Please enter your email address using this format: X@X.com.

**Comment by Mottie** [comment]
Almost all of the inputs have some restriction. For example, the following patterns are allowed per the schema, but may be prevented from being entered by the actual form element:

At least for the HLR form (copied from from 526):

* Rep Name (30 max): a-z, A-Z, 0-9, single quote, period, comma, ampersand, hash (#) and space
* Address line 1 (50 max): a-z, A-Z, 0-9, single quote, period, comma, ampersand, hash (#) and space
* Address lines 2 & 3 (20 max): a-z, A-Z, 0-9, single quote, period, comma, ampersand, hash (#) and space
* City (30 max): a-z, A-Z, 0-9, single quote, number sign and space
* Zip code: 0-9, dash... it allows 2 patterns: 5 digits, or 5 digits-dash-4 digits
* Phone: 10 digits only
* Date, pattern: YYYY-MM-DD
  * YYYY - any 4 digit number, or "XXXX" - the "XXXX" pattern is prevented by the input pattern
  * MM - 1-12, or "XX" - the "XX" is not a select option
  * DD - 1-31, or "XX" - the "XX" is not a select option

Other forms have looser requirements. We'll probably need to address making these consistent in the future:

Form 10-10EZ:

* First Name (1-30): Anything
* Middle: No limitations
* Last Name (2-30): Anything
* Phone: 10 digits only
* Provider insurance (100 max): Anything
* Policy Holder name (50 max): Anything
* Policy number (30 max): Anything
* Insurance Group code (30 max): Anything
* SSN. pattern: 9 digits, or 3 digits dash 2 digits dash 4 digits
